scene/3d/sprite_3d.cpp

```diff
@@ -366,11 +366,17 @@ void Sprite3D::_draw() {
 		final_rect.position * pixel_size,
 
 	};
+
+	// Properly setup UVs for impostor textures (AtlasTexture).
+	RID texture_rid = texture->get_rid();
+	Vector2 src_tsize = Vector2(
+			VS::get_singleton()->texture_get_width(texture_rid),
+			VS::get_singleton()->texture_get_height(texture_rid));
 	Vector2 uvs[4] = {
```

Comment:
Requesting data from VisualServer should never ever happen.

VisualServer may be running on a separate thread, so you will completely block the render thread by doing this.

Simple query Texture instead for the size.